docs/faq/faq.md

```diff
@@ -0,0 +1,236 @@
+
+??? question "How do I reprocess data I already transformed?"
+    Sometimes you need to reprocess data that has already been loaded and transformed. In SQLMesh, you do that with [restatement plans](../concepts/plans.md#restatement-plans).
+
+    Specify the [`plan` command's](../reference/cli.md#plan) `--restate-model` option and the model name(s) you want to reprocess. Applying the plan will reprocess those models and all models downstream from them. You can use the `--start` and `--end` options to limit the reprocessing to a specific date range.
+
+??? question "How do I reuse an existing table instead of creating a new one?"
+    Sometimes a table is too large to completely rebuild for a breaking change, so you need to reuse the existing table. This is done with [forward-only plans](../concepts/plans.md#forward-only-plans). Create one by adding the `--forward-only` option to the [`plan` command]((../reference/cli.md#plan)): `sqlmesh plan [environment name] --forward-only`.
+
+    When a forward-only plan is applied to the `prod` environment, none of the plan's changed models will have new physical tables created for them. Instead, physical tables from previous model versions are reused. All changes made as part of a forward-only plan automatically get a forward-only category assigned to them - they can't be mixed together with regular breaking/non-breaking changes.
+
+    You can retroactively apply the forward-only plan's changes to existing data in the production environment with [`plan`'s `--effective-from` option](../reference/cli.md#plan).
+
+## Databases/Engines
+
+??? question "What databases/engines does SQLMesh work with?"
+    SQLMesh works with BigQuery, Databricks, DuckDB, PostgreSQL, GCP PostgreSQL, Redshift, Snowflake, and Spark. See [this page](../integrations/engines.md) for more information.
+
+??? question "When would you use different databases for executing data transformations and storing state information?"
+    SQLMesh requires storing state information about projects and when their transformations were run. By default, it stores this information in the same database where the models run. 
+    
+    Unlike data transformations, storing state information requires database transactions. Some databases, like BigQuery, aren’t optimized for executing transactions, so storing state information in them can slow down your project. If this occurs, you can store state information in a different database, such as PostgreSQL, that executes transactions more efficiently.
+
+## Warnings and Errors
+
+??? question "Why did I get the warning 'Query cannot be optimized due to missing schema for model [...]'?"
+    SQLMesh uses its knowledge of table schema (column names and data types) to optimize model queries and create column-level lineage. SQLMesh does not have schema knowledge for data sources outside the project and will generate this warning when a model selects from one.
```

Comment:
Should we use a different term instead of "optimize" here? It feels like a misnomer, given that we're not leveraging SQLGlot's full optimizer, but just the subset that canonicalizes the query.

Reply:
Having said that, if people agree maybe we should also update the displayed warnings emitted by the runtime?